### PR TITLE
Testing Center/Nation/Community (connects #94,#95,#96)

### PR DESCRIPTION
**Pull request by Rupesh87**

### CODE REVIEW

File: couchdb-setup.sh [hunk 1]
```diff
@@ -43,6 +43,8 @@ curl -X PUT $COUCHURL/_global_changes
 curl -X PUT $COUCHURL/meetups
 curl -X PUT $COUCHURL/resources
 curl -X PUT $COUCHURL/courses
+curl -X PUT $COUCHURL/nations
+curl -X PUT $COUCHURL/community
```
Comment (lmmrssa) [inline couchdb-setup.sh]: change to communityregistrationrequests

File: package.json [hunk 2]
```diff
@@ -7,12 +7,12 @@
...
     "v-serve": "vagrant ssh -c 'cd /vagrant;ng serve'",
     "build": "ng build",
-    "v-build":"vagrant ssh -c 'cd /vagrant;ng build'",
+    "v-build": "vagrant ssh -c 'cd /vagrant;ng build'",
     "test": "ng test",
     "v-test": "vagrant ssh -c 'cd /vagrant;ng test'",
     "lint": "ng lint",
     "v-lint": "vagrant ssh -c 'cd /vagrant;ng lint'",
-    "v-lint-fix": "vagrant ssh -c 'cd /vagrant;ng lint --fix'",
+    "v-lint-fix":"vagrant ssh -c 'cd /vagrant;ng lint --fix'",
```
Comment (lmmrssa) [inline package.json]: add space

File: src/app/community/community.component.html [hunk 3]
```diff
@@ -0,0 +1,54 @@
+<h2 i18n>Communities List</h2>
+<br/>
+<div class="row">
+  <div class="col-md-3">
+      <input (keyup)="filterCommunity()" class="form-control" [(ngModel)]="filter" placeholder="Search By Nation Name">
```
Comment (lmmrssa) [inline src/app/community/community.component.html]: add translation i18n-placeholder like this

File: src/app/community/community.component.html [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  </div>
+  <div class="col-md-3">
+    <select  [(ngModel)]="selectedValue" (change)="filterCommunity()" class="form-control">
+        <option  value="">All</option>
```
Comment (lmmrssa) [inline src/app/community/community.component.html]: add translation i18n

File: src/app/home/home-router.module.ts [hunk 5]
```diff
@@ -5,13 +5,18 @@ import { DashboardComponent } from '../dashboard/dashboard.component';
...
 import { CoursesComponent } from '../courses/courses.component';
+import { CommunityComponent } from '../community/community.component';
+import { NationComponent } from '../nation/nation.component';
 
 const routes: Routes = [
   { path: '', component: HomeComponent,
     children: [
       { path: '', component: DashboardComponent},
       { path: 'users', component: UsersComponent},
+      { path: 'nation', component: NationComponent},
```
Comment (lmmrssa) [inline src/app/home/home-router.module.ts]: why this code is added twice

File: src/app/home/home.module.ts [hunk 6]
```diff
@@ -21,8 +25,10 @@ import { CourseValidatorService } from '../validators/course-validator.service';
     NavigationComponent,
     UsersComponent,
     CoursesComponent,
-    FormErrorMessagesComponent
+    CommunityComponent,
+    FormErrorMessagesComponent,
+    NationComponent
   ],
-  providers: [CourseValidatorService]
+  providers: [CourseValidatorService,NationValidatorService]
```
Comment (lmmrssa) [inline src/app/home/home.module.ts]: add space

File: src/app/nation/nation.component.html [hunk 7]
```diff
@@ -0,0 +1,38 @@
+<h1>Add Nation </h1>
```
Comment (lmmrssa) [inline src/app/nation/nation.component.html]: no translation added

File: src/app/nation/nation.component.ts [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+
+import { CouchService } from '../shared/couchdb.service';
+import { CustomValidators } from '../validators/custom-validators';
+import { NationValidatorService } from '../validators/nation-validator.service';
+
+@Component({
+	selector: 'app-nation',
+	templateUrl: './nation.component.html',
+	styleUrls: ['./nation.component.scss'],
+	template: `<h1>Nation List</h1>
```
Comment (lmmrssa) [inline src/app/nation/nation.component.ts]: translation not added

File: src/app/nation/nation.component.ts [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+  getNationList() {
+		this.couchService.get('nations/_all_docs?include_docs=true')
+			.then((data) => {
+				this.nation = data.rows;
+			}, (error) => this.message = 'There was a problem getting NationList');
+	}
+
+  onSubmit(nation) {
+    if(nation.nation_name !== "" && nation.nationurl !== "" && nation.type !=="") {
+      this.couchService.post('nations', {'adminname':nation.adminName, 'nationname': nation.name,'nationurl':nation.nationUrl, 'type':nation.type}, )
```
Comment (lmmrssa) [inline src/app/nation/nation.component.ts]: why space not added

File: src/app/validators/nation-validator.service.ts [hunk 10]
```diff
@@ -0,0 +1,43 @@
+import { Injectable } from '@angular/core';
+import { AbstractControl, ValidationErrors } from '@angular/forms';
+// Make sure not to import the entire rxjs library!!!
+import { Observable } from 'rxjs/Observable';
+import { Subject } from 'rxjs/Subject';
+
+import searchDocuments from '../courses/constants';
```
Comment (lmmrssa) [inline src/app/validators/nation-validator.service.ts]: why courses/constants is used?

File: couchdb-setup.sh (hunk 1 above)
Comment (Rupesh87) [reply couchdb-setup.sh]: changed to communityregistrationrequests

File: src/app/home/home-router.module.ts (hunk 5 above)
Comment (Rupesh87) [reply src/app/home/home-router.module.ts]: removed the repeated code.

File: src/app/community/community.component.html [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+        {{community.nationName}}
+      </td>
+      <td>
+        {{community.lastPublicationsSyncDate}}
+      </td>
+      <td>
+        {{community.lastActivitiesSyncDate}}
+      </td>
+      <td>
+        <button i18n class="btn btn-danger" ngMOdel (click)="deleteCommunity(community._id,community._rev)">Delete</button>
```
Comment (lmmrssa) [inline src/app/community/community.component.html]: ngModel

File: src/app/nation/nation.component.ts [hunk 12]
```diff
@@ -0,0 +1,82 @@
...
+    });
+  }
+
+  cancel() {
+    this.location.back();
+  }
+
+  getNationList() {
+    this.i=0;
+		this.couchService.get('nations/_all_docs?include_docs=true')
```
Comment (lmmrssa) [inline src/app/nation/nation.component.ts]: indentation

File: src/app/validators/nation-validator.service.ts [hunk 13]
```diff
@@ -0,0 +1,43 @@
...
+    };
+    // calls service every .5s for input change
+    return Observable.timer(500).switchMap(() => {
+      return this.nationCheckerService$(ac.value).map(res => {
+        if (res) {
+          return errMessage;
+        }
+        return null;
+      });
+    });;
```
Comment (lmmrssa) [inline src/app/validators/nation-validator.service.ts]: two semicolon

**Comment by lmmrssa** [comment]
also there are extra files 'package-lock.json' and 'nation.component.scss'

File: src/app/nation/nation.component.ts [hunk 14]
```diff
@@ -0,0 +1,81 @@
...
+	selector: 'app-nation',
+	templateUrl: './nation.component.html'
+})
+
+export class NationComponent implements OnInit{
+  message = '';
+  nation = [];
+  readonly dbName = 'nations';
+  nationForm : FormGroup;
+  i;
```
Comment (paulbert) [inline src/app/nation/nation.component.ts]: Change to descriptive variable name

**Pull request merged**